Fruit base

ATSchoolENikonov opened the pull request:
2 задание

methshest committed:
5.6
2 задание

methshest committed:
5.6
2 задание правки

Separador commented on 2.58_homework/com/fruitbase/Simulation.java:
```diff
@@ -14,10 +16,52 @@ public static void main(String[] args) {
+            } catch (IOException e) {
+                System.out.println("Ошибка при импорта каталога из " + s);
+                System.exit(12);
+            } catch (UnsupportedClassVersionError e) {
```
Там возникает InvalidClassException

Separador reviewed (changes requested):
1) У тебя одинаковый код для экспорта и импорта из переданного аргумента пробует достать путь до файла.  Давай для этого сделаем отдельный метод `String extractPath(String argument)`в классе Simulation

2) Чтобы метод main оставался читаемым, давай сделаем отдельные методы `tryExport(FruitBase base, String path)` и `tryImport(FruitBase base, String path)`, в которые переместим части по вызову экспорта/импорта и обработки исключений.

methshest committed:
5.6
Переделал методы в мейне+ переставил и заменил исключение

Separador commented on 2.58_homework/com/fruitbase/Simulation.java:
```diff
@@ -4,20 +4,71 @@
+
     public static void main(String[] args) {
         FruitBase fr = new FruitBase();
+        Simulation sim = new Simulation();
```
Мне кажется лучше будет, если методы `tryImport`, `tryExport` и `extractPath` сделать статическими, тк экземпляр создается просто для их вызова.

methshest committed:
6.4
Добавил анонимные класыы, у меня тут получается что 3 кастомер берет цену уже от оставшегося груза, так как два предыдущих разбирают свои покупки и сам груз уменьшается в весе и цене. Возможно так и должно работать

methshest committed:
6.4
Доработал по макс цене, добавив метод в интерфейс по посику макс цены

Separador commented on 2.58_homework/com/fruitbase/Simulation.java:
```diff
@@ -72,7 +74,27 @@ public static void main(String[] args) {
+                    public List<Fruit> takeFruits(Delivery cargo) {
+                        List<Fruit> k = cargo.getFruits();
+                        for (int i = 0; i < cargo.getFruits().size(); i++) {
+                            if ((k.get(i).getPrice().doubleValue() /cargo.getMaxPrice().doubleValue() ) > 0.75) {
```
Лучше `cargo.getMaxPrice()` вызвать до цикла, чтобы на каждую итерацию не добавлять лишние одинаковые вычисления.

methshest committed:
6.4
Вынес за цикл макс цену
